jwk.go

```diff
@@ -240,16 +244,42 @@ func (k *JSONWebKey) UnmarshalJSON(data []byte) (err error) {
 	*k = JSONWebKey{Key: key, KeyID: raw.Kid, Algorithm: raw.Alg, Use: raw.Use, Certificates: certs}
 
 	k.CertificatesURL = raw.X5u
-	k.CertificateThumbprintSHA1 = raw.X5tSHA1.bytes()
-	k.CertificateThumbprintSHA256 = raw.X5tSHA256.bytes()
+
+	// x5t parameters are base64url-encoded SHA thumbprints
+	// See RFC 7517, Section 4.8, https://tools.ietf.org/html/rfc7517#section-4.8
+	x5tSHA1bytes, err := base64.RawURLEncoding.DecodeString(raw.X5tSHA1)
+	if err != nil {
+		return errors.New("square/go-jose: invalid JWK, x5t header has invalid encoding")
+	}
+
+	if len(x5tSHA1bytes) > sha1.Size {
```

Comment:
@mangas we had a bit of a discussion in https://github.com/square/go-jose/pull/303#discussion_r416998839 and I think it would be good to drop a comment here to explain the fallback and how RFC is unclear wether the thumbprints should be raw bytes or a hex string representation. Also if it's a hex then it we know the length -- it would be `2 * sha1.Size` and (and equivalent for SHA256).

Reply:
Also, can we get 2 valid roundtrip tests in which one has `base64(SHA-bytes)` and the other with `base64(SHA-hex)`?

Reply:
We have that test, TestRoundtripX509Hex. Takes the hex and produces a non-hex checksum result.

Reply:
Cool, I'm make the size checks more strict and add a comment.

Reply:
Let me know if this comments is clear enough